packages/alfa-protractor/src/protractor.ts

```diff
@@ -0,0 +1,277 @@
+/// <reference lib="dom" />
+/// <reference lib="dom.iterable" />
+
+import { Device } from "@siteimprove/alfa-device";
+import {
+  Attribute,
+  Block,
+  Comment,
+  Declaration,
+  Document,
+  Element,
+  FontFaceRule,
+  ImportRule,
+  KeyframeRule,
+  KeyframesRule,
+  MediaRule,
+  NamespaceRule,
+  Node,
+  PageRule,
+  Rule,
+  Sheet,
+  StyleRule,
+  SupportsRule,
+  Text,
+  Type,
+} from "@siteimprove/alfa-dom";
+import { Request, Response } from "@siteimprove/alfa-http";
+import { Page } from "@siteimprove/alfa-web";
+
+import { ElementFinder } from "protractor";
+
+export namespace Protractor {
+  export type Type = ElementFinder;
+
+  export async function toPage(value: Type): Promise<Page> {
```

Comment:
I'm seeing this (and companion functions) in most test integration packages, without seeing obvious differences. Should it be refactored?

Reply:
It _should_, yes, I just haven't yet found a good way of sharing that particular bit of functionality. The issue is that the functions need to be defined _within_ the lambda passed to APIs like `#executeScript()` as the lambdas will be stringified and sent over the wire. Stuff like this won't work:

```ts
// packages/alfa-dom/native.ts

export function toNode(node: globalThis.Node): Node;
```

```ts
// packages/alfa-protractor/src/protractor.ts

import { toNode } from "@siteimprove/alfa-dom/native";

const node = browser.executeScript(() => toNode(window.document));
```